subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/connection/NoToolingApiConnection.java

```diff
@@ -64,4 +66,9 @@ public void runTests(TestExecutionRequest testExecutionRequest, ConsumerOperatio
         throw Exceptions.unsupportedFeature(operationParameters.getEntryPointName(), distribution, "2.6");
     }
 
+    @Override
+    public void notifyDaemonsAboutChangedPaths(List<String> changedPaths, ConsumerOperationParameters operationParameters) {
+        throw Exceptions.unsupportedFeature(operationParameters.getEntryPointName(), distribution, "6.1");
```

Comment:
should this fail? if it does, then we have a different contract for target versions of Gradle that support the tooling API but not the notification feature (ignore) and target versions of Gradle that don't support the tooling API at all (fail). On the other hand, there's almost no chance that this method will ever run anywhere except perhaps our test suite.

Reply:
I think it should fail like the other methods. I can make the method no-op, though it seems like the user should be informed early that the version of the distribution is too low.